improver/spotdata/height_adjustment.py

```diff
@@ -0,0 +1,207 @@
+                extract the desired set of coordinates from the neighbour cube.
+        """
+        self.neighbour_selection_method = neighbour_selection_method
+        self.threshold_coord = None
+        self.units = None
+
+    def adjust_prob_cube(self, spot_cube: Cube, vertical_displacement: Cube) -> Cube:
+        """
+        Adjust probability spot forecasts based on the vertical displacement of sites
+        in relation to orography.
+
+        Args:
+            spot_cube:
+                A cube of spot probability forecasts.
+            vertical_displacement:
+                A cube containing information about the difference between spot
+                data site height and the orography grid square height.
+
+        Returns:
+            A cube with the same metadata and shape as spot_cube but with probabilities
+            adjusted to be relative to the site altitude rather than grid square altitude.
+        """
+        coord_list = [c.name() for c in spot_cube.dim_coords]
+        enforce_coordinate_ordering(
+            spot_cube, [self.threshold_coord.name, "spot_index"]
+        )
+
+        thresholds = self.threshold_coord.points
+        spot_index = spot_cube.coord("spot_index").points
+        shape = spot_cube.shape
+
+        broadcast_max = np.transpose(
+            np.broadcast_to(
+                np.amax(spot_cube.data, axis=1), (len(thresholds), spot_cube.shape[0])
+            )
+        )
+        broadcast_min = np.transpose(
+            np.broadcast_to(
+                np.amin(spot_cube.data, axis=1), (len(thresholds), spot_cube.shape[0])
+            )
+        )
+
+        broadcast_thresholds = np.broadcast_to(
+            thresholds, (vertical_displacement.shape[0], len(thresholds))
+        )
+        broadcast_vertical_displacement = np.transpose(
+            np.broadcast_to(
+                vertical_displacement.data,
+                (len(thresholds), vertical_displacement.shape[0]),
+            )
+        )
+        desired_thresholds = broadcast_thresholds + broadcast_vertical_displacement.data
+
+        # creates a list of pairs of values of spot index with the thresholds that need to
+        # be calculated for the spot index
+        coord = list(product(spot_index, thresholds))
+        needed_pair = []
+        for index, threshold in zip(spot_index, desired_thresholds):
+            needed_pair.extend(list(product([index], threshold)))
+
+        # interpolate across the cube and request needed thresholds
+        interp = LinearNDInterpolator(coord, spot_cube.data.flatten())
+        spot_data = np.reshape(interp(needed_pair), shape)
+
+        # Points outside the range of the original data return NAN. These points are replaced
+        # with the highest or lowest along the axis depending on the whether the vertical
+        # displacement was positive or negative
+        indicies = np.where(np.isnan(spot_data))
```

Comment:
```suggestion
        indices= np.where(np.isnan(spot_data))
```
Should also be changed in the next few lines as well

Reply:
Done. One day I'll learn how to spell indices.